disperser/cmd/dataapi/main.go

```diff
@@ -147,3 +160,70 @@ func RunDataApi(ctx *cli.Context) error {
 
 	return err
 }
+
+func getWallet(config Config, ethClient common.EthClient, logger logging.Logger) (walletsdk.Wallet, error) {
+	var wallet walletsdk.Wallet
+	if !config.FireblocksConfig.Disable {
+		validConfigflag := len(config.FireblocksConfig.APIKeyName) > 0 &&
+			len(config.FireblocksConfig.SecretKeyName) > 0 &&
+			len(config.FireblocksConfig.BaseURL) > 0 &&
+			len(config.FireblocksConfig.VaultAccountName) > 0 &&
+			len(config.FireblocksConfig.WalletAddress) > 0 &&
+			len(config.FireblocksConfig.Region) > 0
+		if !validConfigflag {
+			return nil, errors.New("fireblocks config is either invalid or incomplete")
+		}
+		apiKey, err := secretmanager.ReadStringFromSecretManager(context.Background(), config.FireblocksConfig.APIKeyName, config.FireblocksConfig.Region)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read fireblocks api key %s from secret manager: %w", config.FireblocksConfig.APIKeyName, err)
+		}
+		secretKey, err := secretmanager.ReadStringFromSecretManager(context.Background(), config.FireblocksConfig.SecretKeyName, config.FireblocksConfig.Region)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read fireblocks secret key %s from secret manager: %w", config.FireblocksConfig.SecretKeyName, err)
+		}
+		fireblocksClient, err := fireblocks.NewClient(
+			apiKey,
+			[]byte(secretKey),
+			config.FireblocksConfig.BaseURL,
+			config.FireblockAPITimeout,
+			logger.With("component", "FireblocksClient"),
+		)
+		if err != nil {
+			return nil, err
+		}
+		wallet, err = walletsdk.NewFireblocksWallet(fireblocksClient, ethClient, config.FireblocksConfig.VaultAccountName, logger.With("component", "FireblocksWallet"))
+		if err != nil {
+			return nil, err
+		}
+		sender, err := wallet.SenderAddress(context.Background())
+		if err != nil {
+			return nil, err
+		}
+		if sender.Cmp(gethcommon.HexToAddress(config.FireblocksConfig.WalletAddress)) != 0 {
+			return nil, fmt.Errorf("configured wallet address %s does not match derived address %s", config.FireblocksConfig.WalletAddress, sender.Hex())
+		}
+		logger.Info("Initialized Fireblocks wallet", "vaultAccountName", config.FireblocksConfig.VaultAccountName, "address", sender.Hex())
+	} else if len(config.EthClientConfig.PrivateKeyString) > 0 {
+		privateKey, err := crypto.HexToECDSA(config.EthClientConfig.PrivateKeyString)
```

Comment:
wonder if there is any benefit if we put raw private key into the secret manager like the api keys above?

Reply:
Yes. That would be arguably more secure, but it adds a dependency on AWS secret manager so neither wallet will be functional if secret manager is down